pkg/cookies/csrf.go

```diff
@@ -63,7 +66,10 @@ func NewCSRF(opts *options.Cookie) (CSRF, error) {
 
 // LoadCSRFCookie loads a CSRF object from a request's CSRF cookie
 func LoadCSRFCookie(req *http.Request, opts *options.Cookie) (CSRF, error) {
-	cookie, err := req.Cookie(csrfCookieName(opts))
+	
+	// csrfCookieName will include state to have multiple csrf in case of parallel requests
+	state := req.URL.Query()["state"][0][0:csrfStateLength - 1]	
```

Comment:
I'm also curious if we can potentially manage multiple CSRF state checks in an array structure serialized in 1 CSRF struct serialized into the cookie?

Things on my mind:

- The cookie manages the OIDC nonce across requests too. How will that handle multiple (seems fine).
- We encrypt the cookie payload (oauth state & nonce contents). Making the state part of the cookie name will slightly weaken that posture. But I bet it's fine, the encryption was likely overkill and those would only expose a portion of the OAuth state nonce.

Reply:
I'm also not convinced the multiple tab issue was the sole source of these intermittent CSRF failures.

At my previous company where I ran this in production, whenever users reported this error (and when it happened to me) multiple tabs logging in a once was never a symptom.

Reply:
This long running issue has more samples of tips people have (and theories on why this happens): https://github.com/oauth2-proxy/oauth2-proxy/issues/817

Do we think think this will fix some of the prefetch, favicon & JS issues people have uncovered as reasons?

Reply:
Hi @JoelSpeed, Added some more testing for edge cases. 
@NickMeves I'm not sure about the multiple tabs, I have to say that from my experience, issues with multiple tabs can result from session management problems on the provider side.
I do suspect the parallel requests to favicon and prefetch are related to this.